src/selection.js

```diff
@@ -14,33 +14,72 @@ module.exports = Selection;
  * facilities to manipulate, save, restore the
  * selection, across DOM modifications.
  */
-function Selection(nativeSelection) {
+function Selection(nativeSelection, editor) {
   var me = this;
+  me.editor = editor;
 
   var native = me.native = assert(nativeSelection);
 
   me.setCaret = function(point) {
     var pair = Point.check(point).toNodeOffset();
 
-    native.setBaseAndExtent(pair[0], pair[1], pair[0], pair[1]);
+    me.setBaseAndExtent(pair[0], pair[1], pair[0], pair[1]);
   };
 
   me.setEndpoints = function(anchor, focus) {
     var anchorPair = Point.check(anchor).toNodeOffset();
     var focusPair = Point.check(focus).toNodeOffset();
 
-    native.setBaseAndExtent(anchorPair[0], anchorPair[1], focusPair[0], focusPair[1]);
+    me.setBaseAndExtent(anchorPair[0], anchorPair[1], focusPair[0], focusPair[1]);
   };
 
-  /** Convenience function - returns true if the selection is collapsed */
+  me.setBaseAndExtent = function(anchorParent, anchorOffset, focusParent, focusOffset) {
```

Comment:
The documented abstraction level of this class is in terms of Point locations and various high level utilities.

I recommend you split this cross-browser abstraction into a separate interface called "NativeSelection", which basically presents the subset of raw native API that we want to use - e.g. the setBaseAndExtent method & similar.  Then have a helper function that constructs an implementation for the current browser (e.g. it could use if-statements to select what implementation of setBaseAndExtent to add to the object, or alternatively, have different classes e.g. WebkitSelection, FirefoxSelection, and construct the appropriate one).

See the "TODO: implement wrapper to abstract browser variants." commented out section of code below, that was a start on this.

For convenience, can also make the Selection constructor use the above helper by default, so people who just want to get something working don't have to write all the wiring boilerplate.

And i'd make the editor parameter the first parameter, since it's the only essential one. I.e.

// only need to pass in nativeSelection explicitly if you want to do something fancy, or need support for some new api or user agent that QED hasn't supported yet, etc.
function Selection(editor, nativeSelection) {
  nativeSelection = nativeSelection || createNativeSelection();
}